network/network.go

```diff
@@ -26,30 +26,36 @@ import (
 )
 
 type Network struct {
-	Name   string
-	Config *subnet.Config
-
-	sm     subnet.Manager
-	ipMasq bool
-	be     backend.Backend
-	lease  *subnet.Lease
+	Name       string
+	Config     *subnet.Config
+	ctx        context.Context
+	cancelFunc context.CancelFunc
+
+	sm         subnet.Manager
+	ipMasq     bool
+	be         backend.Backend
+	lease      *subnet.Lease
 }
 
-func New(sm subnet.Manager, name string, ipMasq bool) *Network {
+func NewNetwork(sm subnet.Manager, name string, ipMasq bool) *Network {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I recently realized that a slicker way is to wrap ctx with another `WIthCancel` (https://godoc.org/golang.org/x/net/context#WithCancel). This way you get another `cancel` function but the outer one still works. This way you don't have to:

```
<-ctx.Done()
for _, net := range m.networks {
    net.Cancel()
}
```